src/main/java/io/trino/plugin/tiledb/TileDBRecordCursor.java

```diff
@@ -449,11 +464,145 @@ private void setRanges(TileDBSplit split) throws TileDBError
                 dimIdx++;
                 dimension.close();
             }
+            //iterate attributes
+            HashMap<String, Attribute> attributes = arraySchema.getAttributes();
+            Iterator it = attributes.entrySet().iterator();
+            QueryCondition finalQueryCondition = null;
+            while (it.hasNext()) {
+                Map.Entry pair = (Map.Entry) it.next();
+                Attribute att = (Attribute) pair.getValue();
+                Pair attBounds = getBoundsForAttribute(split, att);
+                Object lowBound;
+                Object highBound;
+                boolean isString = att.getType().javaClass().equals(String.class);
+                if ((isString && attBounds.getFirst() != null && attBounds.getFirst().toString().equals(""))) { //empty string case
+                    QueryCondition cond;
+                    if (att.getNullable()) {
+                        cond = new QueryCondition(ctx, att.getName(), "".getBytes(), att.getType().javaClass(), TILEDB_EQ);
+                    }
+                    else {
+                        cond = new QueryCondition(ctx, att.getName(), " ".getBytes(), att.getType().javaClass(), TILEDB_EQ);
+                    }
+                    if (finalQueryCondition == null) {
+                        finalQueryCondition = cond;
+                    }
+                    else {
+                        finalQueryCondition = finalQueryCondition.combine(cond, TILEDB_AND);
+                    }
+                    continue;
+                }
+                if ((attBounds.getFirst() != null || attBounds.getSecond() != null)) {
+                    if (attBounds.getFirst() == null) {
+                        if (isString) {
```

Comment:
Can we move this block and the `else if` block below to a function? It's the same code in both cases, right?

Reply:
The similarity is high but these two blocks of code use slightly different methods. (e.g. attbounds.gitFirst(), attbounds.getSecond())

Reply:
Please try to simplify this a little bit... eg can we write it something like this? Starting from [this line](https://github.com/TileDB-Inc/TileDB-Trino/pull/73/files#diff-a7ef783f2f7566ea09b02ebf4e11d481aae1713e7c8057e6d483cd7242c2f601R534) I think we could use a getter to implement the switches on the bound type/content:

```
//QueryCondition cond1 = new QueryCondition(ctx, att.getName(), lowBound, att.getType().javaClass(), TILEDB_GE);
//QueryCondition cond2 = new QueryCondition(ctx, att.getName(), highBound, att.getType().javaClass(), TILEDB_LE);
QueryCondition cond1 = conditionForBound(attBounds.getFirst());
QueryCondition cond2 = conditionForBound(attBounds.getSecond());
QueryCondition cond3 = cond1.combine(cond2, TILEDB_AND);
```

with the impl something like this:
```
private QueryCondition conditionForBound(Attribute attr, Object bound, tiledb_query_condition_op_t op) {
  boolean isString = att.getType().javaClass().equals(String.class);

  // handle the empty string cases
  if (isString && bound != null && bound.toString().equals(""))) {
    if (att.getNullable()) {
      return new QueryCondition(ctx, att.getName(), "".getBytes(), att.getType().javaClass(), TILEDB_EQ);
    } else {
       return new QueryCondition(ctx, att.getName(), " ".getBytes(), att.getType().javaClass(), TILEDB_EQ);
    }
  }
  if (isString) {
    bound = bound.toString().getBytes(); // not sure if this is legal; use a temporary if not?
  }
  return new QueryCondition(ctx, att.getName(), bound, att.getType().javaClass(), op);
}
```

Reply:
Simplified!

Reply:
Thanks